### DISCUSSION TITLE
How to select DASH audio/video representation before the player is initialised with it?

### DISCUSSION THREAD

**Discussion by benesva4** [videojs/video.js#7407] [Q&A]
Is such thing possible? We face some problems with both - for video it's the problem of UHD video having different codecs than lower resolutions, so the player just stop playing when we switch. We solved this problem by having UHD in a different source and when the user check UHD option (manually), we reset the player and put in the new source. This is bad, because user has no adaptive streaming available anymore.  I guess it could be solved via `html5.vhs.bandwidth` option.
For the sound the problem is even more complicated, since the player starts playing representation with two channels and when we switch to to audiotrack that have representation with six channels, the player downmix 5.1 back to 2.0.
So is there any way to tell the player which representation should it pick at the start? Thank you very much.

I just noticed there is an issue where there is written something about better support for playlist with mixed codecs, so maybe my question is less relevant, but it would be still great to know if there is a way.

**Comment by benesva4** [comment]
And more qustion is there any way to get information from the manifest like the audio configuration/number of channels?

**Reply by gkatsev** [reply]
Assuming it's in the manifest, you can get it from the playlist object's `attributes` property, perhaps something like:
```js
player.tech().vhs.playlists.media().attributes
```
However, I'm not sure that everything is exposed directly.

**Comment by gkatsev** [comment]
Unfortunately, the best way to select which rendition before playback starts is the `bandwidth` option you mentioned.
